libres/lib/enkf/row_scaling.cpp

```diff
@@ -183,54 +182,67 @@ void row_scaling::multiply(matrix_type *A, const matrix_type *X0) const {
     matrix_free(X);
 }
 
-template <typename T> void row_scaling::assign(const T *data, int size) {
-    this->resize(size);
+void RowScaling::assign_vector(const float *data, size_t size) {
+    m_resize(size);
     for (int index = 0; index < size; index++)
-        this->assign(index, data[index]);
+        assign(index, data[index]);
 }
 
-/*
-  Below here is a C api for binding to Python.
-*/
-
-row_scaling_type *row_scaling_alloc() {
-    row_scaling_type *scaling = new row_scaling();
-    return scaling;
+void RowScaling::assign_vector(const double *data, size_t size) {
+    m_resize(size);
+    for (int index = 0; index < size; index++)
+        assign(index, data[index]);
 }
 
-row_scaling_type *row_scaling_alloc_copy(const row_scaling_type *scaling) {
-    return new row_scaling(*scaling);
+namespace {
+void setitem(RowScaling &self, int index, double value) {
+    self.assign(index, value);
 }
 
-void row_scaling_free(row_scaling_type *scaling) { delete scaling; }
+double getitem(const RowScaling &self, int index) { return self[index]; }
 
-double row_scaling_iget(const row_scaling_type *scaling, int index) {
-    return scaling->operator[](index);
-}
+const auto assign_vector_doc = R"(
+Assign tapering value for all elements via a vector.
 
-double row_scaling_iset(row_scaling_type *scaling, int index, double value) {
-    return scaling->assign(index, value);
-}
+The assign_vector() function will resize the row_scaling vector to the
+number of elements in the scaling_vector, and assign a value to all
+elements.
 
-double row_scaling_clamp(const row_scaling_type *scaling, double value) {
-    return scaling->clamp(value);
-}
+A typical situation might be to load the scaling data as a EclKW
+instance from a grdecl formatted file. Before the assign_vector() can
+be called we must transform to an only active elements representation
+and use a numpy view:
 
-void row_scaling_multiply(const row_scaling_type *scaling, matrix_type *A,
-                          const matrix_type *X0) {
-    scaling->multiply(A, X0);
-}
+    # Load scaling vector from grdecl file; typically created with
+    # geomodelling software.
+    with open("scaling.grdecl") as fh:
+        kw_global = EclKW.read_grdecl(fh, "SCALING")
 
-int row_scaling_get_size(const row_scaling_type *row_scaling) {
-    return row_scaling->size();
-}
+    # Create a ecl_kw copy with only the active elements.
+    kw_active = grid.compressed_kw_copy(kw_global)
```

Comment:
What is `grid` in this example?

Reply:
Probably `EclGrid`: https://github.com/equinor/ecl/blob/main/python/ecl/grid/ecl_grid.py